include/net/mqtt_helper.h

```diff
@@ -45,8 +45,22 @@ struct mqtt_helper_buf {
 	size_t size;
 };
 
-typedef void (*mqtt_helper_handler_t)(struct mqtt_evt *evt);
-typedef void (*mqtt_helper_on_connack_t)(enum mqtt_conn_return_code return_code);
+/** @brief Handler invoked for events that are receieved from the MQTT stack.
+ *	   This callback handler can be used to filter incoming MQTT events before they are
+ *	   processed by the MQTT helper library.
+ *
+ *  @param client Pointer to the MQTT client instance.
+ *  @param event Pointer to the MQTT event.
+ *
+ *  @retval 0 if the event is handled by the caller. No further processing of the MQTT event will
+ *	      be carried out by the MQTT helper library.
+ *  @retval 1 if the MQTT helper library should continue to process the event after
+ *	      the handler returns.
+ */
+typedef bool (*mqtt_helper_on_event_t)(struct mqtt_client *const client,
```

Comment:
Maybe we could call it something like `_on_all_events` just to make it super clear when it's called

Reply:
yes